genet/schedule_elements.py

```diff
@@ -1927,3 +2112,14 @@ def verify_graph_schema(graph):
                 missing_attribs = required_service_attributes - set(service_dict.keys())
                 raise ScheduleElementGraphSchemaError(f'Service {service_id} is missing the following attributes: '
                                                       f'{missing_attribs}')
+
+
+def read_vehicle_types(yml):
+    """
+    :param yml: path to .yml file based on example vehicles config in `genet/configs/vehicles/vehicle_definitions.yml`
+        or a bytes stream of that file
+    :return:
+    """
+    if persistence.is_yml(yml):
+        yml = io.open(yml, mode='r')
+    return yaml.load(yml, Loader=yaml.FullLoader)['VEHICLE_TYPES']
```

Comment:
`yml` can also be an already opened bytes stream of a yml file, so the file extension check is checking that we're given a file rather than the stream. What would be a better way of doing this?